src/main/java/org/wildfly/security/credential/PasswordCredential.java

```diff
@@ -33,16 +34,29 @@
  * A credential for password authentication.
  */
 public final class PasswordCredential implements AlgorithmCredential {
+    private final Supplier<Provider[]> providers;
     private final Password password;
 
     /**
      * Construct a new instance.
      *
      * @param password the password (must not be {@code null})
+     * @param providers the Security Providers to use when locating the PasswordFactory
      */
-    public PasswordCredential(final Password password) {
+    public PasswordCredential(final Password password, final Supplier<Provider[]> providers) {
```

Comment:
Will have another look and see if there is a way to pull this out.  The problem was the verify method that gets called doesn't take a Provider supplier so was hard coded to Security.getProviders().